packages/manager/apps/web/client/app/domain/anycast/domain-dns-anycast.controller.js

```diff
@@ -1,79 +1,91 @@
+import find from 'lodash/find';
+import get from 'lodash/get';
+import {
+  DNS_ANYCAST_SERVICE_TYPE,
+  DNS_ANYCAST_PLANCODE,
+  DNS_ANYCAST_PRICING_MODE,
+  DNS_ANYCAST_DURATION,
+} from './domain-dns-anycast.constants';
+
 export default class DomainDnsAnycastActivateCtrl {
   /* @ngInject */
-  constructor($translate, $window, Alerter, atInternet, Domain) {
-    this.$translate = $translate;
-    this.$window = $window;
-    this.Alerter = Alerter;
-    this.atInternet = atInternet;
-    this.Domain = Domain;
+  constructor($q, WucOrderCartService, SessionService) {
+    this.$q = $q;
+    this.WucOrderCartService = WucOrderCartService;
+    this.SessionService = SessionService;
   }
 
   $onInit() {
-    this.domainId = this.domainName;
-    this.optionName = 'dnsAnycast';
-    this.loading = false;
-    this.loadOptionDetails();
-  }
-
-  loadOptionDetails() {
     this.loading = true;
-    this.error = null;
-
-    return this.Domain.getOptionDetails(this.domainId, this.optionName)
-      .then((data) => {
-        this.optionDetails = data;
-      })
-      .catch(() => {
-        this.error = this.$translate.instant(
-          'domain_configuration_dnsanycast_fail',
+    return this.fetchCatalogOffer(this.domainName)
+      .then((offer) => {
+        this.offer = offer;
+        this.offerPrice = find(this.offer.prices, {
```

Comment:
Only one price available ?

Reply:
yes there is only one price, i preferred to use a find to be more explicit and showing that the price used is P1Y (one year subscription) ; instead of simply picking the first result